eng/templates/stages/mirror.yml

```diff
@@ -160,7 +160,7 @@ stages:
         #   083b8b9fe09d04205e87bca0b378e4a3ad74a239        refs/tags/v8.0.0-rc.1.23419.4
         #   113d797bc90104bb4f1cc51e1a462cf3d4ef18fc        refs/tags/v8.0.0-rc.1.23419.4^{}
         # We want to match on both.
-        tagGrepQuery="refs/tags/$tag_name\(^{}\)\?$"
+        tagGrepQuery="refs/tags/$tag_name\(\^{}\)\?$"
 
         echo "git ls-remote --tags destination | grep -q $tagGrepQuery"
```

Comment:
Sorry for the random drive by comment: if you ever see SIGPIPE or `broken pipe` errors, these `grep -q` calls might be the culprit. `grep -q` closes the input pipe as soon as it finds a match, so if the first command (`git`) is still trying to send some input, it will exit with that broken pipe error.

Reply:
Thank you for the pointer! Our validation builds caught this issue almost instantly and your comment save us a good chunk of time for investigating the issue.

Tried to fix it in https://github.com/dotnet/source-build/pull/3731